src/telemetry.rs

```diff
@@ -35,10 +43,68 @@ pub fn setup_logging() {
 #[cfg(not(feature = "console"))]
 pub fn setup_logging() {
     Lazy::force(&APPLICATION_START_TIME);
-    let filter_layer = tracing_subscriber::EnvFilter::try_from_default_env()
+    let filter = tracing_subscriber::EnvFilter::try_from_default_env()
         .or_else(|_| tracing_subscriber::EnvFilter::try_new("info"))
         .unwrap();
-    tracing_subscriber::registry()
-        .with(tracing_subscriber::fmt::layer().with_filter(filter_layer))
-        .init();
+    let (filter_layer, reload_handle) =
+        reload::Layer::new(tracing_subscriber::fmt::layer().with_filter(filter));
+    unsafe {
+        PRI_LOG_HANDLE = Some(Box::leak(Box::new(LogHandle(reload_handle))));
```

Comment:
can we avoid unsafe here?

Reply:
> can a Cell be used?

let me check on it and get back to you, thanks

Reply:
avoided the "unsafe", please review